src/web/mjs/connectors/Cuutruyen.mjs

```diff
@@ -0,0 +1,152 @@
+            let excute =
+        cuudrmJs1 +
+        cuudrmJs2 +
+        cuudrmJs3 +
+        cuudrmJs4 +
+        cuudrmJs5 +
+        fastestSmallestTextEncoderDecoderJs +
+        `return cuudrm.render_image(void 0, void 0, "${drmData}");`;
+            let result = new Function(excute)();
+            return result;
+        }
+        const uri = new URL('/api/v2/chapters/' + chapter.id, this.api);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchJSON(request);
+        const pages = data.data.pages;
+        // Store image url in to array for later use
+        let urlImage = [];
+        pages.forEach((image) => {
+            urlImage.push(image.image_url);
+        });
+        // Store drmData in to array for later use
+        let drmData = [];
+        pages.forEach((item) => {
+            drmData.push(item.drm_data.replace(/\n/g, '\\n'));
+        });
+        // Store decryptedData in to array for later use
+        let decryptedData = [];
+        drmData.forEach(item => {
+            decryptedData.push(imageDecode(item));
+        });
+        let images = [];
+        for (let i = 0; i < pages.length; i++) {
+            images.push(await execute(urlImage[i], JSON.stringify(decryptedData[i])));
+        }
+        async function execute(url, drm) {
+            let response = await fetch(url);
+            if (response) {
+                let blob = await response.blob();
+                let bitmap = await createImageBitmap(blob);
+                let imgWidth = bitmap.width;
+                let imgHeight = bitmap.height;
+                let canvas = document.createElement('canvas');
+                canvas.width = imgWidth;
+                canvas.height = imgHeight;
+                let ctx = canvas.getContext('2d');
+                JSON.parse(drm).forEach((part) => {
+                    let sx = part[0];
+                    let sy = part[1];
+                    let sHeight = part[3];
+                    let dx = part[4];
+                    let dy = part[5];
+                    let dHeight = part[7];
+                    ctx.drawImage(
+                        bitmap,
+                        sx,
+                        sy,
+                        imgWidth,
+                        sHeight,
+                        dx,
+                        dy,
+                        imgWidth,
+                        dHeight
+                    );
+                });
+                let image = await new Promise((resolve) => {
+                    let imageRecompressionFormat = Engine.Settings.recompressionFormat.value;
+                    let imageRecompressionQuality = Engine.Settings.recompressionQuality.value;
+                    canvas.toBlob((blob) => {
+                        resolve(URL.createObjectURL(blob));
```

Comment:
Memory leak warning : you use createObjectURL but you NEVER use revokeObjectURL().